### ISSUE TITLE
Design  and write code for the machine learning model

### ISSUE THREAD

**Issue by alecswift** [alecswift/safe-path-finder#6]
Machine Learning Model Description:

The Tabular Machine Learning model will predict the likelihood of an accident using weather, location, road conditions, and light conditions. Considering the data is structured in a table, I will use ensembles of decision trees as the machine learning technique. Dependent variable: Likelihood of an accident. Independent variables: Location, Weather, Road conditions, Light conditions, day of the week.

Libraries: 

Create a panda Dataframe from the excel dataset file and convert all categorical variables to numeric data using pandas Categorical class.
Separate the data into a training set and a validation set
Decide on the parameters for the decision tree such as the maximum depth and the splitter (best or random).
Create a decision tree from the tabular data using the scikit-learn library
Throughout the training process use a metric function such as the root mean squared error of the actual values and the predicted values to measure performance.
Use the bagging technique, create a large number of decision trees each trained on a different subset of the dataset and take the average of their predictions.
Find the confidence of the predictions using a method such as the standard deviation of predictions among the trees. This will give us more information than the binary outcome of whether or not an accident occurred.
Measure the relative importance of the independent variables on the model predictions and simplify the model by removing variables that have a small effect on the predictions.
Analyze the partial dependency of variables in the model
Fiddle around with the model by implementing and testing other techniques such as boosting and embeddings.
If there is time, build and train a neural network for comparison with the random forest model.
I can take it a step further and create an ensemble by averaging the predictions of both the random forest model and the neural network model.
